backend/src/apiserver/webhook/pipelineversion_webhook.go

```diff
@@ -120,6 +121,12 @@ func (p *PipelineVersionsWebhook) ValidateDelete(_ context.Context, _ runtime.Ob
 }
 
 func (p *PipelineVersionsWebhook) Default(ctx context.Context, obj runtime.Object) error {
+
+	return p.DefaultWithRetry(ctx, obj, 0)
+}
+
+// DefaultWithRetry will attempt to run the logic again in case the cache was not updated yet.
+func (p *PipelineVersionsWebhook) DefaultWithRetry(ctx context.Context, obj runtime.Object, tries int) error {
```

Comment:
Let's just use a for loop over the `p.Client.Get` instead of recursion. It's less duplicative and easier to read.